src/main/java/com/google/devtools/build/lib/bazel/commands/ModCommand.java

```diff
@@ -595,6 +611,55 @@ private BlazeCommandResult runTidy(CommandEnvironment env, BazelModTidyValue mod
 
     return reportAndCreateTidyResult(env, modTidyValue);
   }
+  
+  private static ImmutableList<String> compareOutputWithFiles(CommandEnvironment env, BazelModTidyValue modTidyValue, String out) {
+    ImmutableList<String> filesNeedingFormat = ImmutableList.of();
+    Path rootString = env.getWorkspace();
+    for (PathFragment moduleFilePath : modTidyValue.moduleFilePaths()) {
+      Path fullPath = rootString.getRelative(moduleFilePath);
+      if (fullPath.exists()) {
+        try {
+          String contents = FileSystemUtils.readContent(fullPath, UTF_8);
+          if (!out.contains(contents)) {
+            filesNeedingFormat = ImmutableList.<String>builder().addAll(filesNeedingFormat).add(moduleFilePath.getPathString()).build();
+          }
+        } catch (IOException e) {
+          env.getReporter().handle(Event.error("Failed to read file: " + fullPath));
+        }
+      }
+    }
+    return filesNeedingFormat;
+  }
+
+  private static BlazeCommandResult reportAndCreateTidyDryRunResult(
+    CommandEnvironment env, BazelModTidyValue modTidyValue, String out) {
+    ImmutableList<String> filesNeedingFormat = compareOutputWithFiles(env, modTidyValue, out);
+    if (modTidyValue.fixups().isEmpty() && filesNeedingFormat.isEmpty()) {
+      return BlazeCommandResult.success();
+    } else {
+      String lintErrors = "";
+      if (!modTidyValue.fixups().isEmpty()) {
+        lintErrors += String.format("Files with errors:\n%s",
+          modTidyValue.fixups().stream()
+            .map(fixup -> {
+              String extensionId = 
+                fixup.usage().getExtensionBzlFile() + "%" + fixup.usage().getExtensionName();
+              return fixup.usage().getProxies().stream()
+                .map(p -> String.format("  %s: %s", p.getLocation().toString(), extensionId))
+                .collect(joining("\n"));
```

Comment:
Instead of collecting here and then collecting again in the outer stream, use `flatMap`.